lib/mongo_ecto.ex

```diff
@@ -407,23 +407,68 @@ defmodule Mongo.Ecto do
   end
 
   @doc false
-  def load(:binary_id, data),
-    do: Ecto.Type.load(ObjectID, data, &load/2)
-  def load(Ecto.Date, {date, _time}),
-    do: Ecto.Type.load(Ecto.Date, date, &dump/2)
-  def load(:map, []),
-    do: {:ok, %{}}
+  def load(_type, nil),
+    do: {:ok, nil}
+  # wat
+  def load(:binary_id, <<_::binary-12>> = binary),
+    do: {:ok, ObjectID.encode(binary)}
```

Comment:
If you return the whole ObjectId in the linked code above, you should be able to remove this clause.